### PR TITLE
Tests: Add integration tests with core blocks schema validation

### LINKED ISSUE #36319: Block attributes: Consider making `type` a required property
## What problem does this address?
Currently block attributes requires either a `type` or an `enum`.

Only adding a `enum` on a attribute causes some issues on SSR blocks. When the attributes is prepared for render they are [validated with `rest_validate_value_from_schema`](https://github.com/WordPress/wordpress-develop/blob/31e92478728c8a95e8e297847699b03842558cd3/src/wp-includes/class-wp-block-type.php#L310). `rest_validate_value_from_schema` requires a `type` to be set. Without it a notice will be thrown because `type` doesn't exist (and a `_doing_it_wrong`).

Ex on block that will fail:
```php
<?php
register_block_type(
    'core/example',
    array(
        'attributes' => array(
            'value' => array(
                'enum' => array( 'ok', false ),
            ),
        ),
        'render_callback' => function( $block_attributes ) {
            return $block_attributes['value'];
        },
    )
);
```

Not sure there is any use cases where you can't set a `type`. The example above would be solved by adding `type => array( 'string', 'boolean' )`.

## What is your proposed solution?
In attributes make `type` required.

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Follow up for #35900.
Closes #36319.

Adds a missing `$schema` field to a few more blocks. I also wrote integration tests to enforce that every block has `block.json` with the `$schema` field and the metadata validates against the local schema.

## How has this been tested?

```sh
npm run test-unit test/integration/blocks-schema.test.js
```

## Types of changes

Automated tests added.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: test/integration/blocks-schema.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import Ajv from 'ajv-draft-04';
```
Comment (gziolo) [inline test/integration/blocks-schema.test.js:4]: I had to install `ajv-draft-04` to make everything work with the defined schema: 
http://json-schema.org/draft-04/schema#

> Out of the box, Ajv supports JSON Schema (drafts 04, 06, 07, 2019-09 and 2020-12) and JSON Type Definition (RFC8927 (opens new window)).

However, in practice, only JSON Schema draft-06/07/2019-09/2020-12 standards work with the core `ajv` package. Is there any reason we picked `draft-04`?

Comment (fabiankaegy) [reply test/integration/blocks-schema.test.js:4]: @gziolo This is all the context I have. We briefly talked about it in the initial review of the `block.json` introduction PR. 

https://github.com/SchemaStore/schemastore/pull/1879#issuecomment-944426503

**Review by fabiankaegy — COMMENTED** [review]
Outside of the outstanding comment regarding the Schema version I think this looks great and does what we need :)

File: test/integration/blocks-schema.test.js (hunk 1 above)
Comment (gziolo) [reply test/integration/blocks-schema.test.js:4]: I guess it’s fine to leave it as is. Thank you for sharing the context 💯

**Review by mkaz — APPROVED** [review]
This is great! 🌟 

Thanks for catching the blocks I missed, were they new? Good idea adding a test to make sure blocks have the file also. 👍 

Validation script works well, I tested by making an invalid block.json file both by having improper JSON, and incorrect values that don't match the schema. The script picked up on all issues.

Great addition! 🚀

File: test/integration/blocks-schema.test.js (hunk 1 above)
Comment (ajlende) [reply test/integration/blocks-schema.test.js:4]: SchemaStore has some best practices listed in their contributing docs that I was also basing my decisions off of. https://github.com/SchemaStore/schemastore/blob/master/CONTRIBUTING.md#best-practices

**Comment by gziolo** [comment]
> Thanks for catching the blocks I missed, were they new?

I know that some of them were new.

**Pull request merged**